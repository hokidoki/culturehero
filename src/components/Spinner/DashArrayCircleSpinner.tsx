import React from 'react';
import type DashArrayCircleSpinnerProps from './DashArrayCircleSpinner.types';
import DashArrayCircleSpinnerStyle from './DashArrayCircleSpinner.styles';
import SpinnerLayoutStyle from './SpinnerLayout.styles';

export default function DashArrayCircleSpinner({}: DashArrayCircleSpinnerProps) {
  return (
    <DashArrayCircleSpinnerStyle.SVG viewBox="0 0 100 100">
      <DashArrayCircleSpinnerStyle.Circle cx={'50'} cy={'50'} r={'48'} />
    </DashArrayCircleSpinnerStyle.SVG>
  );
}

export function DashArrayCircleSpinnerWithLayout<T>({
  layoutProps,
}: {
  layoutProps?: Parameters<(typeof SpinnerLayoutStyle)['ParentCoverLayout']>[0];
}) {
  return (
    <SpinnerLayoutStyle.ParentCoverLayout {...layoutProps}>
      <DashArrayCircleSpinner />
    </SpinnerLayoutStyle.ParentCoverLayout>
  );
}
