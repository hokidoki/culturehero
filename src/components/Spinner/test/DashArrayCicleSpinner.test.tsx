import { render } from '@render';
import DashArrayCircleSpinner from '../DashArrayCircleSpinner';

describe('CommentListItem', () => {
  it('match snapshot', () => {
    const utils = render(<DashArrayCircleSpinner />);
    expect(utils.container).toMatchSnapshot();
  });
});
