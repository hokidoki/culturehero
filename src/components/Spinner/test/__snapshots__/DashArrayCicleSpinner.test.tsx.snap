// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentListItem match snapshot 1`] = `
.c1 {
  width: 100%;
  height: 100%;
  fill: transparent;
  stroke: white;
  stroke-width: 2px;
  transform-origin: center;
  stroke-dasharray: 301.59289474462014;
  animation: ieKHan 2s infinite ease-in-out alternate;
}

.c0 {
  width: 100px;
  height: 100px;
  max-width: 100%;
  max-height: 100%;
  margin: 0 auto;
}

<div>
  <svg
    class="c0"
    viewBox="0 0 100 100"
  >
    <circle
      class="c1"
      cx="50"
      cy="50"
      r="48"
    />
  </svg>
</div>
`;
